edb/pgsql/resolver/expr.py

```diff
@@ -75,7 +75,20 @@ def infer_alias(res_target: pgast.ResTarget) -> Optional[str]:
 def resolve_ResTarget(
     res_target: pgast.ResTarget,
     *,
-    existing_names: Optional[Set[str]] = None,
+    existing_names: Set[str],
+    ctx: Context,
+) -> Tuple[Sequence[pgast.ResTarget], Sequence[context.Column]]:
+    targets, columns = _resolve_ResTarget(
+        res_target, existing_names=existing_names, ctx=ctx
+    )
+
+    return (targets, columns)
+
+
+def _resolve_ResTarget(
```

Comment:
Why this extra level of indirection?

Reply:
Initially i thought I was gonna do validation after the call of the inner function, but I've later realized that validation depends on data I discard during resolution, so I had to do validation during resolution.

So this is a leftover.